R/get_timescale_data.R

```diff
@@ -0,0 +1,107 @@
+#' Get geological timescale data
+#'
+#' This function takes a name of a geological timescale and returns data for the
+#' timescale. Valid names include those of built-in `data.frames` ([GTS2020()]
+#' and [GTS2012()]) and exact matches to those hosted by Macrostrat (see full
+#' list here: <https://macrostrat.org/api/defs/timescales?all>). If a built-in
+#' `data.frame` is specified with `name`, a specific `rank` may also be
+#' specified. Valid ranks include "period", "epoch", "stage", "era", and "eon"
+#' as well as partial matches of those names (e.g., "per" or "st"). If a rank
+#' is supplied, only the intervals of that rank will be returned from the
+#' specified timescale.
+#'
+#' @param name \code{character}. The name of the desired timescale.
+#' @param rank \code{character}. The name of the desired rank (if `name`
+#'   specifies a built-in timescale).
+#' @return A `data.frame` with at least the following columns:
+#'   \item{name}{the names of the time intervals.}
+#'   \item{max_age}{the oldest boundaries of the time intervals, in millions of
+#'     years.}
+#'   \item{min_age}{the youngest boundaries of the time intervals, in millions
+#'     of years.}
+#'   \item{abbr}{either traditional abbreviations of the names of the time
+#'     intervals (if they exist) or custom abbreviations created with R.}
+#'   \item{color}{hex color codes associated with the time intervals (if
+#'     applicable).}
+#' @importFrom utils read.csv
+#' @importFrom curl nslookup
+#' @importFrom grDevices col2rgb
+#' @export
+get_timescale_data <- function(name = "GTS2020", rank = NULL) {
+  possible_names <- c("GTS2020", "GTS2012")
+  name_match <- charmatch(name, possible_names)
+  if (!is.na(name_match)) {
+    if (name_match == 0) {
+      stop("'name' matches both built-in scales. Please be more specific.",
```

Comment:
```suggestion
      stop("`name` matches both built-in scales. Please be more specific.",
```
Elsewhere we use backticks for arguments